`now dev` 0.12.x on Windows fails with Cannot find module

Title says it all. My app was working for me with `now dev` on Monday, but updates released since then broke any routes based on TypeScript files that are nested within folders.

For example, if `http://localhost:3000/api-broken` is configured to point to `zeit/api/api-handler.ts`, this is the output I get:

```
> GET /api-broken
> Building @now/node@canary:zeit/api/api-handler.ts
downloading user files...
download complete [1ms]
installing dependencies for user's code...
Found `engines` in `package.json`, selecting range: 10.x
installing to C:\Users\Tyler Church\Desktop\vanilla-from-wsl\zeit\api
npm WARN vanilla@1.0.0 No repository field.

audited 2 packages in 0.375s
found 0 vulnerabilities

install complete [4083ms]
running user script...
script complete [3ms]
tracing input files...
Using TypeScript 3.5.2 (now internal)
trace complete [1219ms]
> Built @now/node@canary:zeit/api/api-handler.ts [6s]
Cannot find module './zeit/api/api-handler.ts'
1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.
Did you forget to add it to "dependencies" in `package.json`?
```

If I move the relevant `.ts` file to the root folder and update `now.json` to reflect this, things work. But obviously this is not ideal from a code organization standpoint.

The problem seems to have something to do with the built `.ts` file not getting copied or renamed or something. When the `.ts` file is in the root folder, the build output looks like this:

![image](https://user-images.githubusercontent.com/42798/62251801-c1911f80-b3a5-11e9-804a-522b9704842f.png)

But when I've got it in a folder, it looks like this:

![image](https://user-images.githubusercontent.com/42798/62251813-cc4bb480-b3a5-11e9-9935-d16450fb31ab.png)

Notice that the `.ts` file is missing in the second image, but a `.js` file is there in its place.

I'm running:

- Windows 10 v1903
- Node.js v10.16.0
- Now CLI v15.8.7
- NPM v6.9.0
- Yarn v1.17.3
- TypeScript v3.5.3

A repo that reproduces the issue is here: https://github.com/tylerc/zeit-now-ts-debugging

That app running is available here: https://zeit-now-ts-debugging.tylerc.now.sh

Luckily it seems to be working when deployed, but it makes development difficult on Windows machines.

I suspect that something related to this is the cause of this problem: https://github.com/zeit/now-builders/issues/855 I had to update my `now.json` to use `@now/node@canary` to fixed the error mentioned in that issue. But after doing so I ran into this new error.

If there's anything you'd like me to try, or any more information I can provide, please let me know.

Seeing the same "Cannot find module" error in my logs using

  - `@now/node` and
  - `@now/node@canary`

but not `@now/node@0.11.1`

> Cannot find module './http-proxy/'
> Did you forget to add it to "dependencies" in `package.json`?

I can't be sure (because the logs don't give line numbers for debug), but I think this is the line causing the problem in my case: https://github.com/http-party/node-http-proxy/blob/7e6c66a7e485a6c0ec3a1c567bbe800fdc56c9fd/lib/http-proxy.js#L4

That module is being required by another module required in my code.

@styfle Unfortunately when I use `@now/node@canary`. it still does not seem to be working for me in either my test repo or in the app where I originally encountered the issue.

The output from `now dev` is nearly identical to when I first reported the issue, though that "if you want to compile the package/file..." message is now gone:

```
> GET /api-broken
> Building @now/node@canary:zeit/api/api-handler.ts
downloading user files...
download complete [1ms]
installing dependencies for user's code...
Found `engines` in `package.json`, selecting range: 10.x
installing to C:\Users\Tyler Church\Desktop\fresh-repo\zeit\api
npm WARN vanilla@1.0.0 No repository field.

audited 2 packages in 0.407s
found 0 vulnerabilities

install complete [4011ms]
running user script...
script complete [3ms]
tracing input files...
Using TypeScript 3.5.2 (now internal)
trace complete [1239ms]
> Built @now/node@canary:zeit/api/api-handler.ts [6s]
Cannot find module './zeit/api/api-handler.ts'
Did you forget to add it to "dependencies" in `package.json`?
```

--- a/packages/now-node/src/index.ts
+++ b/packages/now-node/src/index.ts
@@ -239,7 +239,12 @@ async function compile(
       }
     }
     // Rename .ts -> .js (except for entry)
-    if (path !== entrypoint && tsCompiled.has(path)) {
+    // There is a bug on Windows where entrypoint uses forward slashes
+    // and workPath uses backslashes so we use resolve before comparing.
+    if (
+      resolve(workPath, path) !== resolve(workPath, entrypoint) &&
+      tsCompiled.has(path)
+    ) {
       preparedFiles[
         path.slice(0, -3 - Number(path.endsWith('x'))) + '.js'
       ] = entry;

